dashboard/app/helpers/levels_helper.rb

```diff
@@ -47,6 +47,11 @@ def set_videos_and_blocks_and_callouts_and_instructions
 
     if current_user && params[:load_previous]
       @start_blocks = current_user.last_attempt(@level).try(:level_source).try(:data)
+    elsif current_user && params[:level_source_id]
+      @level_source = LevelSource.find(params[:level_source_id])
```

Comment:
I think we should be doing this (and the above) on the client side. It's here for now because I tried to do that and blockly freaked out when I tried to load the workspace blocks twice. I'll add a TODO saying this